state-chain/pallets/cf-validator/src/lib.rs

```diff
@@ -1543,35 +1532,20 @@ impl<T: Config> Pallet<T> {
 		}
 		log::info!(target: "cf-validator", "Starting rotation");
 
-		let qualified_bidders = Self::get_qualified_bidders::<T::KeygenQualification>();
-
-		let delegation_snapshot = Self::build_delegation_snapshot(
-			qualified_bidders.clone().into_iter().map(|bid| bid.bidder_id.into()).collect(),
-		);
+		let delegation_snapshots = Self::build_delegation_snapshots::<T::KeygenQualification>();
 
-		let get_avg_bid_for_validator_if_managed =
-			|validator_id: T::AccountId| -> Option<T::Amount> {
-				let operator_id = ManagedValidators::<T>::get(validator_id)?;
-				let snap = delegation_snapshot.get(&operator_id)?;
-				Some(snap.avg_bid)
-			};
+		let auction_bids = delegation_snapshots
+			.values()
+			.flat_map(|snapshot| snapshot.effective_validator_bids())
+			.map(|(bidder_id, amount)| Bid { bidder_id, amount })
+			.collect::<Vec<_>>();
```

Comment:
As an example of what I mean - see how it simplified generation of the auction bids here.